app/jobs/reactions/create_job.rb

```diff
@@ -0,0 +1,19 @@
+module Reactions
+  class CreateJob < ApplicationJob
+    queue_as :reaction_create
+
+    def perform(user_id:, reactable_id:, reactable_type:, category:)
+      user = User.find_by(id: user_id)
+      reactable = reactable_type.constantize.find_by(id: reactable_id)
```

Comment:
I suggest adding a check before `constantize` to be sure that we constantize a valid class.
It could be a guard clause `return unless %w(Acticle Comment).include?(reactable_type)` or raising an exception if the `reactable_type` is not in thos list

Reply:
Would it be better to use [safe_constantize](https://apidock.com/rails/String/safe_constantize) instead, because it returns ``nil`` if no class is found? Something like:

```ruby
reactable = reactable_type.safe_constantize&.find_by(id: reactable_id)
```

I'm slightly concerned ``%w(Article Comment)`` might not be an extensive list with all reactable classes in the future (not sure I need to worry about that).

Reply:
It's not only about the class could not exist, but it could be another class (not a valid `reactable`) which could cause weird behavior. I'm not also a fan of using `&` when it's possible to avoid it (it's not a "confident" code if we are constantly checking for `nil`)
Imo explicit checking also works as documentation, that's why I suggested it. In the future, it would be required to add the classes to the list (if needed) and add the corresponding tests.